packages/lab/src/stepped-tracker/SteppedTracker.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  ComponentPropsWithoutRef,
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  Children,
+  useState,
+  isValidElement,
+  useEffect,
+  createContext,
+  useContext,
+  useMemo,
+} from "react";
+import { clsx } from "clsx";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { makePrefixer } from "@salt-ds/core";
+
+import steppedTrackerCss from "./SteppedTracker.css";
+
+import useDetectTruncatedText, {
+  GetOverflowRef,
+} from "./useDetectTruncatedText";
+
+export type SteppedTrackerContextType = {
+  activeStep: number;
+  getOverflowRef: GetOverflowRef;
+  hasTooltips: boolean;
+  totalSteps: number;
+  isWithinSteppedTracker: boolean;
+};
+
+const defaultSteppedTrackerContext = {
+  activeStep: 0,
+  hasTooltips: false,
+  totalSteps: 1,
+  getOverflowRef: () => undefined,
+  isWithinSteppedTracker: false,
+};
+
+const SteppedTrackerContext = createContext(
+  defaultSteppedTrackerContext as unknown as SteppedTrackerContextType
+);
+
+export const useSteppedTrackerContext = () => useContext(SteppedTrackerContext);
+
+type TrackerStepNumberContextType = number;
+
+const TrackerStepContext = createContext<TrackerStepNumberContextType>(0);
+
+export const useTrackerStepContext = () => useContext(TrackerStepContext);
+
+const withBaseName = makePrefixer("saltSteppedTracker");
+
+export interface SteppedTrackerProps extends ComponentPropsWithoutRef<"ul"> {
+  /**
+   * The index of the current activeStep
+   */
+  activeStep: number;
+  /**
+   * Should be one or more TrackerStep components
+   */
+  children: ReactNode;
+}
+
+const useCheckInvalidChildren = (children: ReactNode) => {
+  useEffect(() => {
+    if (process.env.NODE_ENV !== "production") {
```

Comment:
what's the reason behind this check?

Reply:
Do you mean the hook or the `process.env.NODE_ENV !== "production"`?

The hook is to alert the consumer if they're doing something they shouldn't. E.g. adding random components within the `SteppedTracker` which would break it. Not sure how useful it is since I didn't want to disallow all other React components in case the consumer created a 'WrappedTrackerStep' with the same API which could be valid.

the condition here is so the end-user doesn't get those warnings in a production app.